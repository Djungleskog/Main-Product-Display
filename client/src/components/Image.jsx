import React from 'react';
import PropTypes from 'prop-types';

var Image = function(props) {
  let handleClick = (event) => {
    props.addCarousel(event);
  };
  return (
    <div className='imageContainer'>
      <img className='image' onClick={handleClick} src={props.imageUrl} alt={props.imageAlt}></img>
    </div>
  );
};

Image.propTypes = {
  imageUrl: PropTypes.string,
  imageAlt: PropTypes.string,
  addCarousel: PropTypes.func,
};


export default Image;