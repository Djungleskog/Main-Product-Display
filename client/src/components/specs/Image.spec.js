import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import Image from '../Image.jsx';

describe('Image', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<Image />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});