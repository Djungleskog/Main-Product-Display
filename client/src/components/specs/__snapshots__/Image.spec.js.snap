// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image snapshot renders 1`] = `
<div
  className="imageContainer"
>
  <img
    className="image"
    onClick={[Function]}
  />
</div>
`;
